src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -9820,213 +9820,2124 @@ public void AlwaysDefault_06_1()
+public struct Buffer10
+{
+    private int _element0;
+
+    public Buffer10()
+    {
+        this[0] = 1;
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput: "1 0 1 0", verify: Verification.Fails).VerifyDiagnostics();
+
+            verifier.VerifyIL("Buffer10..ctor",
+@"
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  .locals init (System.Span<int> V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  initobj    ""Buffer10""
+  IL_0007:  ldarg.0
+  IL_0008:  ldc.i4.s   10
+  IL_000a:  call       ""InlineArrayAsSpan<Buffer10, int>(ref Buffer10, int)""
+  IL_000f:  stloc.0
+  IL_0010:  ldloca.s   V_0
+  IL_0012:  ldc.i4.0
+  IL_0013:  call       ""ref int System.Span<int>.this[int].get""
+  IL_0018:  ldc.i4.1
+  IL_0019:  stind.i4
+  IL_001a:  ret
+}
+");
+        }
+
+        [Fact]
+        public void DefiniteAssignment_34()
+        {
+            var src = @"
+class Program
+{
+    static void M()
+    {
+        Buffer2<int> f;
+        f._element0 = 1;
+        _ = f;
+    }
+}
+
+[System.Runtime.CompilerServices.InlineArray(2)]
+public struct Buffer2<T>
+{
+    public T _element0;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            comp.VerifyDiagnostics(
+                // (8,13): error CS0165: Use of unassigned local variable 'f'
+                //         _ = f;
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "f").WithArguments("f").WithLocation(8, 13)
+                );
+        }
+
+        [Fact]
+        public void DefiniteAssignment_35()
+        {
+            var src = @"
+class Program
+{
+    static void M()
+    {
+        Buffer1<int> f;
+        f._element0 = 1;
+        _ = f;
+    }
+}
+
+[System.Runtime.CompilerServices.InlineArray(1)]
+public struct Buffer1<T>
+{
+    public T _element0;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DefiniteAssignment_36()
+        {
+            var src = @"
+class Program
+{
+    static void M()
+    {
+        Buffer2<int> f;
+        f = default;
+        _ = f;
+    }
+}
+
+[System.Runtime.CompilerServices.InlineArray(2)]
+public struct Buffer2<T>
+{
+    public T _element0;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DefiniteAssignment_38()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(2)]
+public struct Buffer2
+{
+    public int _element0;
+
+    public Buffer2()
+    {
+        _element0 = 0;
+        this[0] = 1;
+        this[1] = 2;
```

Comment:
Do these statements affect definite assignment?

Reply:
> Do these statements affect definite assignment?

Assigning to an element with non-zero index doesn't affect the assigned state for the inline array.